packages/amplify-cli/src/extensions/amplify-helpers/update-backend-config.js

```diff
@@ -15,6 +15,25 @@ function updateBackendConfigAfterResourceAdd(category, resourceName, options) {
   }
 }
 
+function updateBackendConfigDependsOn(category, resourceName, value) {
+  const backendConfigFilePath = pathManager.getBackendConfigFilePath();
+  const backendConfig = JSON.parse(fs.readFileSync(backendConfigFilePath));
+
+  if (!backendConfig[category]) {
+    backendConfig[category] = {};
+    backendConfig[category][resourceName] = {};
+  } else if (!backendConfig[category][resourceName]) {
+    backendConfig[category][resourceName] = {};
+  }
+  if (!backendConfig[category][resourceName].dependsOn) {
+    backendConfig[category][resourceName].dependsOn = [];
+  }
+  backendConfig[category][resourceName].dependsOn = value;
```

Comment:
When meta data file is updated, it's Object.assign, and here it's complete replacement? 
if you are going to assign `value` to the `dependOn` property, you don't need to check its existence, the above `if` statement is not needed.

Reply:
The value has all the dependsOn properties. Can remove the check.